audittools/trail.go

```diff
@@ -38,21 +39,15 @@ type AuditTrail struct {
 // a specific RabbitMQ Connection using the specified amqp URI and queue name.
 // The OnSuccessfulPublish and OnFailedPublish closures are executed as per
 // their respective case.
-func (t AuditTrail) Commit(rabbitmqURI, rabbitmqQueueName string) {
-	rc := &RabbitConnection{}
-	connect := func() {
-		if rc == nil || !rc.IsConnected {
-			var err error
-			rc, err = NewRabbitConnection(rabbitmqURI, rabbitmqQueueName)
-			if err != nil {
-				logg.Error(err.Error())
-			}
-		}
+func (t AuditTrail) Commit(rabbitmqQueueName string, rabbitmqURI amqp.URI) {
+	uriStr := rabbitmqURI.String()
+	rc, err := NewRabbitConnection(uriStr, rabbitmqQueueName)
```

Comment:
we already store `rabbitmqQueueName` inside the `RabbitConnection` struct, shall we store a `uriStr` as well and avoid passing these arguments to `refresh` and `refreshIfClosedOrOld` methods?
If you decide to convert methods above to funcs, you can ignore this comment.

Reply:
If the `RabbitConnection` is nil, we won't we able to use these stored values to get a new `RabbitConnection`.

I store the queue name in `RabbitConnection` after the connection has been created and queue is declared and this value is only used in `PublishEvent` which returns prematurely in case of `RabbitConnection == nil`.

We could, however, make `queueName` a param for `PublishEvent`. I am fine either way.